### PR TITLE
fix: add challenge v2 api to avoid piece hash http header too large

### PR DESCRIPTION
**Pull request by will-2012**
### Description

Add challenge v2 API to avoid piece hash HTTP header too large.

### Rationale

Components such as nginx can disconnect because the response header is too large, and users receive a 502 HTTP status code.

### Example

N/A.

### Changes

Notable changes: 
* Gateway add new challenge API v2.

### CODE REVIEW

File: modular/gater/admin_handler.go [hunk 1]
```diff
@@ -377,6 +378,169 @@ func (g *GateModular) getChallengeInfoHandler(w http.ResponseWriter, r *http.Req
 	log.CtxDebug(reqCtx.Context(), "succeed to get challenge info")
 }
 
+// getChallengeInfoV2Handler handles get challenge piece info request. Current only greenfield
+// validator can challenge piece is store correctly. The challenge piece info includes:
```
Comment (sysvm) [inline modular/gater/admin_handler.go:382]: Currently, only greenfield validator can challenge piece whether is stored correctly.

File: modular/gater/admin_handler.go [hunk 2]
```diff
@@ -377,6 +378,169 @@ func (g *GateModular) getChallengeInfoHandler(w http.ResponseWriter, r *http.Req
...
+// the challenged piece data, all piece hashes and the integrity hash. The challenger
+// can verify the info whether are correct by comparing with the greenfield info.
+// The difference between the v2 interface and the old interface is the format of the returned results,
```
Comment (sysvm) [inline modular/gater/admin_handler.go:385]: Maybe `the old interface(v1)` is better.

File: modular/gater/admin_handler.go (hunk 1 above)
Comment (will-2012) [reply modular/gater/admin_handler.go]: Fixed.

File: modular/gater/admin_handler.go (hunk 2 above)
Comment (will-2012) [reply modular/gater/admin_handler.go]: Fixed.

**Review by sysvm — APPROVED** [review]
LGTM

**Review by forcodedancing — APPROVED** [review]

**Pull request merged**